hydroflow/examples/two_pc/coordinator.rs

```diff
@@ -172,24 +222,30 @@ pub(crate) async fn run_coordinator(opts: Opts, subordinates: Vec<String>) {
 
     // Phase 2 Response: collect subordinate acks until we have them all
     // Then write to local log and send End message to all subordinates.
-    let mut ack_ctr = HashMap::new();
-    hf.add_subgraph(
+    hf.add_subgraph_stratified(
         "collect acks and end",
+        1,
         p2_recv_pull
             .flatten()
             .map(|msg| {
                 println!("{:?}", msg);
                 msg
             })
-            .filter_map(move |msg| {
-                *ack_ctr.entry((msg.xid, msg.mtype)).or_insert(0) += 1;
-                // HACK
-                if ack_ctr.get(&(msg.xid, MsgType::AckP2)).unwrap_or(&0) + 0 == 3 {
-                    Some(msg)
-                } else {
-                    None
-                }
+            .map_scan(HashMap::<u16, (u32, SubordResponse)>::new(), |txs, msg| {
```

Comment:
similar to above: add comment and/or better variable name to clarify this is an ack counter per transaction.